Tasks/CargoAuthenticateV0/cargoauthenticatemain.ts

```diff
@@ -51,6 +59,45 @@ async function main(): Promise<void> {
             return undefined;
         });
 
+#if WIF
+        const registries = tl.getInput("registryNames");
+        const entraWifServiceConnectionName = tl.getInput("workloadIdentityServiceConnection");
+
+        if (entraWifServiceConnectionName && registries) {
+            const registryNames = registries.split(',');
+            const token = await getFederatedWorkloadIdentityCredentials(entraWifServiceConnectionName);
+
+            if (!token) {
+                throw new Error(tl.loc("FailedToGetServiceConnectionAuth", entraWifServiceConnectionName));
+            }
+
+            for (let registryName of registryNames) {
+                if(!result.registries[registryName]) {
+                    throw new Error(tl.loc("RegistryNotFound", registryName));
+                }
+            }
+
+            const registriesInfo = (<any>Object).entries(result.registries).filter(registry => registryNames.includes(registry[0]));
+            for (let registryInfo of registriesInfo)
+            {
+                var registryUrlStr = url.parse(registryInfo[1].index.replace("sparse+", "")).href;
+                const [registryUrl, tokenName, credProviderName, connectionType] = setRegistryVars(registryUrlStr, registryInfo[0]);
+                if (isValidRegistry(registryUrl, collectionHosts, connectionType))
+                {
+                    tl.warning(tl.loc('ConnectionAlreadySetOverwriting', registryInfo[0], connectionType));
+                    setSecretEnvVariable(tokenName, `Basic ${base64.encode(utf8.encode(`WifBuild:${token}`))}`);
```

Comment:
instead of wifbuild do we have service connection name?

Reply:
It looks like we do have it available from line 64. It may be a good idea to use that instead of WifBuild here